Tools/Scripts/smart-pointer-tool

```diff
@@ -0,0 +1,116 @@
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import argparse
+
+EXPECTATIONS_PATH = '../../Source/{project}/SmartPointerExpectations/{checker_type}Expectations'
+CHECKERS = ['NoUncountedMemberChecker', 'RefCntblBaseVirtualDtor', 'UncountedCallArgsChecker', 'UncountedLocalVarsChecker']
+PROJECTS = ['WebCore', 'WebKit']
+
+
+def parser():
+    parser = argparse.ArgumentParser(description='Various tools for working on smart pointers')
+    parser.add_argument(
+        '--update-expectations', '-u',
+        dest='unexpected_results',
+        default=None,
+        help='Path to unexpected results file, such as WebCore-unexpected-passes.txt. Removes unexpected passes and adds unexpected failures to the expectations files in SmartPointerExpectations.'
+    )
+    parser.add_argument(
+        '--find-expectations', '-f',
+        dest='expected_file',
+        default=None,
+        help='Check if the given file has expected failures'
+    )
+    return parser.parse_args()
+
+
+def modify_expectations_for_checker(checker_type, unexpected_contents, project, type):
+    path_to_expectations = os.path.join(os.path.dirname(__file__), EXPECTATIONS_PATH.format(project=project, checker_type=checker_type))
+    with open(path_to_expectations) as expectations_file:
+        expectations = expectations_file.readlines()
+        prev_len = len(expectations)
+    for line in unexpected_contents:
+        if '=>' in line:
+            new_checker_type = line.split()[-1]
+            modify_expectations_for_checker(new_checker_type, unexpected_contents, project, type)
+        elif line.strip():
+            if type == 'passes':
+                expectations.remove(line)
+            elif line not in expectations:
+                expectations.append(line)
+            else:
+                print(f'{line.strip()} already in {checker_type}Expectations!\n')
+    with open(path_to_expectations, 'w') as expectations_file:
+        expectations_file.writelines(sorted(expectations))
+    print(f'Updated expectations for {checker_type}!')
+    if type == 'passes':
+        print(f'Removed {prev_len - len(expectations)} fixed files.\n')
+    else:
+        print(f'Added {len(expectations) - prev_len} files with issues.\n')
+
+
+def update_expectations(unexpected_results):
+    filename = unexpected_results.split('/')[-1]
+    project, type = filename.split('-')[0], filename.split('-')[-1].removesuffix('.txt')
```

Comment:
It's a bit subtle that the local filename needs to conform to a specific format. Maybe this could be mentioned in the help for `--update-expectations`?